src/main/java/edu/iis/powp/command/gui/ControlsManagerWindow.java

```diff
@@ -118,21 +118,29 @@ public ControlsManagerWindow(Application application) {
     private void moveDownWindow() {
         newPlotter = new MoveDownPlotterDecorator(application.getDriverManager().getCurrentPlotter());
         application.getDriverManager().setCurrentPlotter(newPlotter);
+        DrawerFeature.getDrawerController().clearPanel();
+        commandhistory();
     }
 
     private void moveUpWindow() {
         newPlotter = new MoveUpPlotterDecorator(application.getDriverManager().getCurrentPlotter());
         application.getDriverManager().setCurrentPlotter(newPlotter);
+        DrawerFeature.getDrawerController().clearPanel();
+        commandhistory();
     }
 
     private void moveLeftWindow() {
         newPlotter = new MoveLeftPlotterDecorator(application.getDriverManager().getCurrentPlotter());
         application.getDriverManager().setCurrentPlotter(newPlotter);
+        DrawerFeature.getDrawerController().clearPanel();
+        commandhistory();
     }
 
     private void moveRightWindow() {
         newPlotter = new MoveRightPlotterDecorator(application.getDriverManager().getCurrentPlotter());
```

Comment:
Strange behavior of move decorators when rotation is earlier applied. Maybe the order of transformations is incorrect?!